influxdb3_processing_engine/src/plugins.rs

```diff
@@ -301,13 +363,40 @@ mod python_plugin {
                         };
 
                         runner.advance_time();
-                        if run_in_dedicated_task {
+                        if self.trigger_definition.trigger_settings.run_async {
                             let trigger =self.clone();
                             let fut = async move {ScheduleTriggerRunner::run_at_time(trigger, trigger_time, schema).await};
                             futures.push(fut);
-                        } else if let Err(err) = ScheduleTriggerRunner::run_at_time(self.clone(), trigger_time, schema).await {
-                            self.logger.log(LogLevel::Error, format!("error running scheduled plugin: {}", err));
-                            error!(?self.trigger_definition, "error running scheduled plugin: {}", err);
+                        } else {
+                            match ScheduleTriggerRunner::run_at_time(self.clone(), trigger_time, schema).await {
+                            Ok(plugin_state) => {
```

Comment:
nit: the indentation starting here is subtly off in a few places, probably the `tokio::select!` macro throwing off cargo fmt.